cdp_scrapers/instances/seattle.py

```diff
@@ -0,0 +1,359 @@
+                )
+            )
+
+        return minutes
+
+
+    @staticmethod
+    def legistar_ev_date_time(ev_date: str, ev_time: str) -> datetime.datetime:
+        '''
+        helper func combine legistar ev date and time into datetime
+        '''
+        # 2021-07-09T00:00:00
+        d = datetime.datetime.strptime(ev_date, '%Y-%m-%dT%H:%M:%S')
+        # 9:30 AM
+        t = datetime.datetime.strptime(ev_time, '%I:%M %p')
+        return datetime.datetime(
+            year = d.year, month = d.month, day = d.day,
+            hour = t.hour, minute = t.minute, second = t.second
+        )
+
+
+    def get_events(self,
+        # for the past 2 days
+        begin: datetime.time = datetime.datetime.utcnow() - datetime.timedelta(days = 2),
+        end:   datetime.time = datetime.datetime.utcnow()
+    ) -> List[EventIngestionModel]:
+        '''
+        main getter to retrieve legistar data as cdp ingestion model items
+        '''
+        evs = []
+
+        for legistar_ev in get_legistar_events_for_timespan(
+            self.client_name,
+            begin = begin,
+            end = end,
+            # NOTE: use the range below to retrieve events with 1 containing video
+            #begin = datetime.datetime(year = 2021, month = 7, day = 9),
+            #end = datetime.datetime(year = 2021, month = 7, day = 10),
+        ):
+            session_time = self.legistar_ev_date_time(legistar_ev[LEGISTAR_SESSION_DATE], legistar_ev[LEGISTAR_SESSION_TIME])
+            sessions = []
+
+            # TODO: Session per video_uri/caption_uri ok?
+            for uri in self.get_video_uris(legistar_ev):
+                sessions.append(
+                    Session(
+                        session_datetime = session_time,
+                        session_index = len(sessions),
+                        video_uri = uri[CDP_VIDEO_URI],
+                        caption_uri = uri[CDP_CAPTION_URI],
+                    )
+                )
+            else:
+                # found 0 videos
+                sessions.append(
+                    Session(
+                        session_datetime = session_time,
+                        session_index = 0,
+                        video_uri = None,
+                    )
+                )
+
+            evs.append(
+                EventIngestionModel(
+                    body = Body(name = legistar_ev[LEGISTAR_BODY_NAME]),
+                    event_minutes_items = self.get_event_minutes(legistar_ev[LEGISTAR_EV_ITEMS]),
+                    sessions = sessions,
+                )
+            )
+
+        return evs
+
+
+    def get_video_uris(self, legistar_ev: Dict) -> List[Dict]:
+        '''
+        return url for videos and captions if found in data set from legistar api
+
+        returned data is like
+        [{'video_uri' : 'https://video.mp4', 'caption_uri' : 'https://caption.vtt'}, ...]
+        '''
+        return []
```

Comment:
I would raise not-implemented error here instead of returning empty list. Unless I am missing something.

Reply:
28961327d78004945eaf022949c0f38a6f4d299b